tests/FSharp.Compiler.UnitTests/FsiTests.fs

```diff
@@ -240,4 +244,173 @@ let ``Values are successfully shadowed even with intermediate interactions`` ()
 
     Assert.AreEqual("x", boundValue.Name)
     Assert.AreEqual((1, 2), boundValue.Value.ReflectionValue)
-    Assert.AreEqual(typeof<int * int>, boundValue.Value.ReflectionType)
\ No newline at end of file
+    Assert.AreEqual(typeof<int * int>, boundValue.Value.ReflectionType)
+
+[<Test>]
```

Comment:
I need to add another test to see what the behavior is when we try create a bound value with the name of 'it'.